editoast/src/client/postgres_config.rs

```diff
@@ -25,14 +29,38 @@ pub struct PostgresConfig {
 }
 
 impl PostgresConfig {
-    pub fn url(&self) -> String {
-        format!(
-            "postgresql://{}:{}@{}:{}/{}",
-            self.psql_username,
-            self.psql_password,
-            self.psql_host,
-            self.psql_port,
-            self.psql_database
-        )
+    pub fn url(&self) -> Result<Url> {
+        let base_url = format!("postgresql://{}", self.psql_host);
+        let mut url = Url::parse(&base_url).map_err(|_| PostgresConfigError::Host {
+            value: self.psql_host.clone(),
+        })?;
+        url.set_port(Some(self.psql_port))
+            .map_err(|_| PostgresConfigError::Port {
+                value: self.psql_port,
+            })?;
+        url.set_path(&self.psql_database);
+        url.set_username(&self.psql_username)
+            .map_err(|_| PostgresConfigError::Username)?;
+        url.set_password(Some(&self.psql_password))
+            .map_err(|_| PostgresConfigError::Password)?;
+
+        Ok(url)
     }
 }
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "postgres")]
```

Comment:
This is to avoid setting the status to `500` for each error.

```suggestion
#[editoast_error(base_id = "postgres", default_status = 500)]
```